### PR TITLE
enhancement: Enable tls by default  for `papertrail` and `datadog_logs` sinks

### LINKED ISSUE #2445: Review all sinks to ensure tls settings are set correctly
Looks like with a lot of the movement that we've had with tls types and http clients over the past month or so has caused some confusion around when tls settings are enabled or not by default. We should do a re-review of our sinks and if some sink should have tls settings enabled by default.

### PR DESCRIPTION
**Pull request by ktff**
Closes #2445

This PR does:
* Enables tls by default for `datadog_logs` sink.
* Adds tls support for `papertrail` sink and enables it by default.
* Updates documentation.

### CODE REVIEW

File: .meta/_partials/fields/_tls_connector_options.toml.erb [hunk 1]
```diff
@@ -10,7 +10,7 @@ description = "Configures the TLS options for connections from this sink."
 [<%= namespace %>.tls.children.enabled]
 type = "bool"
 common = true
-default = false
+default = <%= can_enable_default %>
```
Comment (bruceg) [inline .meta/_partials/fields/_tls_connector_options.toml.erb:13]: If I'm reading this right, I'd just name this `enabled_default`. Either that, or make `can_enable` work as a tristate, where unset drops the `enabled` section, and `true` or `false` set the default.

**Review by bruceg — CHANGES_REQUESTED** [review]
Looks like there's a clippy lint to fix, otherwise this looks ok.

File: .meta/_partials/fields/_tls_connector_options.toml.erb (hunk 1 above)
Comment (ktff) [reply .meta/_partials/fields/_tls_connector_options.toml.erb:13]: I wanted use `can_enable_*` format of the name to make it clearer that it's an field under `can_enabled`, but it seems that `can` is getting misinterpreted. 

Tristate would differ from all of the other settings in a sneaky way, as in:
```
  can_enable: false,
  can_verify_certificate: false,
```
would mean `can_verify_certificate` section isn't enabled, but `can_enable` section is enabled, and yet both of them have `false` value.

`enabled_default` is fine so let's go with that.

**Review by bruceg — APPROVED** [review]

**Comment by ktff** [comment]
@binarylogic these documentation changes should be ok so I'll go ahead with the merge.

**Pull request merged**